pkg/platform/model/inventory.go

```diff
@@ -50,25 +52,59 @@ func IngredientByNameAndVersion(language, name, version string, prefix Namespace
 		return nil, locale.NewInputError("inventory_ingredient_not_available", "The ingredient {{.V0}} is not available on the ActiveState Platform", name)
 	}
 
+	var candidates []*IngredientAndVersion
 	for _, ingredient := range results {
 		for _, feature := range ingredient.LatestVersion.ProvidedFeatures {
-			if feature.Feature == nil || *feature.Feature != name {
+			if feature.Feature == nil || strings.ToLower(*feature.Feature) != strings.ToLower(name) {
 				continue
 			}
 
 			for _, ver := range ingredient.Versions {
 				if ver.Version == version {
-					return &IngredientAndVersion{
-						ingredient.V1SearchIngredientsResponseIngredientsItems,
-						ver.Version,
-						ingredient.Namespace,
-					}, nil
+					candidates = append(
+						candidates,
+						&IngredientAndVersion{
+							ingredient.V1SearchIngredientsResponseIngredientsItems,
+							ver.Version,
+							ingredient.Namespace,
+						})
 				}
 			}
 		}
 	}
 
-	return nil, locale.NewInputError("inventory_ingredient_version_not_available", "Version {{.V0}} is not available for package {{.V1}} on the ActiveState Platform", version, name)
+	bestMatch, err := FilterForBestIngredientMatch(candidates, name)
+	if err != nil {
+		return nil, locale.NewInputError("inventory_ingredient_version_not_available", "Version {{.V0}} is not available for package {{.V1}} on the ActiveState Platform", version, name)
```

Comment:
The error from `FilterForBestIngredientMatch` is being thrown away here. Shouldn't we just return the error, or wrap again?

Reply:
thanks for catching that.